Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,421 @@
+            }
+
+            const string peopleMessage = " had the most money ever. ";
+            const string singlePersonMessage = " had the most money ever. ";
+
+            var resultMessage = new StringBuilder();
+            var peopleList = new string[peopleAndBalances.Length];
+            var highestBalanceList = new decimal[peopleAndBalances.Length];
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                highestBalanceList[i] = HighestBalanceForSinglePerson(peopleAndBalances[i]);
+                peopleList[i] = ReturnNameForSinglePerson(peopleAndBalances[i]);
+            }
+
+            var highestBalanceIndex = FindDecimalArrayMax(highestBalanceList);
+            var highestBalance = highestBalanceList[highestBalanceIndex];
+            var personWithHighestBalance = peopleList[highestBalanceIndex];
+
+            if (ArrayElementsAreEqual(highestBalanceList))
+            {
+                resultMessage
+                    .Append(FormatStringAndCommas(peopleList))
+                    .Append(peopleMessage)
+                    .Append(FormatCurrency(highestBalance, currencySymbol))
+                    .Append(messageEnd);
+                return resultMessage.ToString();
+            }
+            else
+            {
+                resultMessage.Clear();
+
+                resultMessage
+                    .Append(personWithHighestBalance)
+                    .Append(singlePersonMessage)
+                    .Append(FormatCurrency(highestBalance, currencySymbol))
+                    .Append(messageEnd);
+
+                return resultMessage.ToString();
+            }
+        }
+
+        /// <summary>
+        /// Return name and current money of the most poor person.
+        /// </summary>
+        public static string FindMostPoorPerson(string[] peopleAndBalances)
+        {
+            if (!ArrayIsValid(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+
+            const string peopleMessage = " have the least money. ";
+            const string singlePersonMessage = " has the least money. ";
+
+            var resultMessage = new StringBuilder();
+            var mostPoorBalanceList = new decimal[peopleAndBalances.Length];
+            var mostPoorPeopleList = new string[peopleAndBalances.Length];
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                mostPoorBalanceList[i] = CurrentBalanceForSinglePerson(peopleAndBalances[i]);
+                mostPoorPeopleList[i] = ReturnNameForSinglePerson(peopleAndBalances[i]);
+            }
+
+            var mostPoorPersonsIndex = FindDecimalArrayMin(mostPoorBalanceList);
+            var mostPoortPerson = mostPoorPeopleList[mostPoorPersonsIndex];
+            var mostPoorPersonsMoney = mostPoorBalanceList[mostPoorPersonsIndex];
+
+            if (mostPoorBalanceList.Length > 2 && ArrayElementsAreEqual(mostPoorBalanceList))
```

Comment:
Instead of magic numbers use named constants

Reply:
I fixed it in commit [529d3c2](https://github.com/csinn/CSharp-From-Zero-To-Hero/pull/161/commits/529d3c25f3eb08ddb62057f8abd449b799f368f7).
Thank you!